instance/tests/integration/factories/instance.py

```diff
@@ -46,8 +46,9 @@ def create(cls, *args, **kwargs):
         # random value for 'internal_lms_domain' if neither 'sub_domain' nor 'internal_lms_domain' are provided.
         if 'sub_domain' not in kwargs and 'internal_lms_domain' not in kwargs:
             kwargs = kwargs.copy()
-            random_id = str(uuid.uuid4())[:8]
```

Comment:
Why these changes were required?

Reply:
This is referring to the last point in the PR description.
The thing is that we are adding a random string prefix to each database and username to avoid name clashing. Currently Ocim uses an [8 char] (https://github.com/open-craft/opencraft/blob/master/instance/tests/integration/factories/instance.py#L49) random [prefix](https://github.com/open-craft/opencraft/blob/master/instance/models/mixins/domain_names.py#L105) for [databases](https://github.com/open-craft/opencraft/blob/master/instance/models/mixins/openedx_database.py#L170) and a [6 char](https://github.com/open-craft/opencraft/blob/master/instance/models/mixins/database.py#L194) random [prefix](https://github.com/open-craft/opencraft/blob/master/instance/models/mixins/openedx_database.py#L204) for users and both of these are different. There is also a cleanup script that will delete every database that's 8 hours or older in case the Ocim integration test failed to delete the databases created. The issue is that the cleanup script can't delete the users bound to a specific database because there's no way to know it (well, I guess we could check permissions but that can get a bit tricky). With this what we do is that we use the same 6 char prefix to create the database and the [users](https://github.com/open-craft/opencraft/blob/josuebc/BB-839/instance/models/openedx_instance.py#L124). By sharing the same prefix we can later get all the old databases and use the same prefix to [query](https://github.com/open-craft/opencraft/blob/josuebc/BB-839/cleanup_utils/mysql_cleanup.py#L270) the username that are bound to that database. Basically creating a namespace.
The other update that I added was to create an empty table called `opencraft` when [creating a Database](https://github.com/open-craft/opencraft/blob/josuebc/BB-839/instance/models/mixins/database.py#L95). That's because we are using `information_schema.tables` to get when databases were last updated but if the integration test didn't create any tables then the database will not show up there. By creating an empty table every time we create a database we make sure every database can be queried through `information_schema.table`

Reply:
Perfect! Nice way of making the cleanup easier. :grin: